st2api/st2api/controllers/resource.py

```diff
@@ -171,92 +187,87 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
             else:
                 filters['__'.join(v.split('.'))] = filter_value
 
-        extra = {
-            'filters': filters,
-            'sort': sort,
-            'offset': offset,
-            'limit': limit
-        }
-        LOG.info('GET all %s with filters=%s' % (pecan.request.path, filters), extra=extra)
-
         instances = self.access.query(exclude_fields=exclude_fields, **filters)
         if limit == 1:
             # Perform the filtering on the DB side
             instances = instances.limit(limit)
 
-        if limit:
-            pecan.response.headers['X-Limit'] = str(limit)
-        pecan.response.headers['X-Total-Count'] = str(instances.count())
-
         from_model_kwargs = from_model_kwargs or {}
-        from_model_kwargs.update(self._get_from_model_kwargs_for_request(request=pecan.request))
+        from_model_kwargs.update(self._get_from_model_kwargs_for_request(**kwargs))
 
         result = []
         for instance in instances[offset:eop]:
             item = self.model.from_model(instance, **from_model_kwargs)
             result.append(item)
 
-        return result
+        resp = Response(json=result)
+        resp.headers['X-Total-Count'] = str(instances.count())
+        if limit:
+            resp.headers['X-Limit'] = str(limit)
+
+        return resp
 
-    def _get_one(self, id, exclude_fields=None):
+    def _get_one(self, id, exclude_fields=None, **kwargs):
```

Comment:
`**kwargs` is a direct equivalent for `...rest` in es6. It means "everything that was defined in spec other than that I've described previously". The benefit of openapi is that it strictly defines the list of arguments your controller would get and there will be no surprises, no additional stuff router puts there on his own initative.

With this in mind, if you don't intend to use some of the arguments inside this method and instead are planning to pass it deeper into the stack, there's really no reason for you to be explicit there and write down every single parameter just so you can write them again in a next line and then change them on every spec change. You just pass `**kwarg` and let underlying function to pick what it needs out of the bucket.